### ISSUE TITLE
Allow group names from FileGroupProvider to be provided in QueryAccessControl

### ISSUE THREAD

**Issue by aakashnand** [trinodb/trino#7730]
Currently, the QueryAccessControl does not provide a group field, this can be helpful for a scenario where users want to create an admin group and allow this group to have `execute`, `kill` `view` permission. 
QueryAccessControl.java relevant lines
https://github.com/trinodb/trino/blob/master/lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/security/QueryAccessRule.java#L40-L41

Error when I tried this
```
Unrecognized field "group" (class io.trino.plugin.base.security.QueryAccessRule), not marked as ignorable (2 known properties: "allow", "user"])
```

If this is a good idea, I would like to take this

**Comment by dain** [comment]
Yes pleas add this.  Also I think this is missing from `SystemInformationRule` also
